### PR TITLE
[Remote Store] Add support for Remote Translog Store upload stats in _nodes/stats/ API

### PR DESCRIPTION
**Pull request by BhumikaSaini-Amazon**
…_nodes/stats/ API (#8908)

---------

### Description
Backport #8908

### Related Issues
#8311 

### Check List
- [x] New functionality includes testing.
  - [ ] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by ashking94 — APPROVED** [review]
Looks good.

File: server/src/main/java/org/opensearch/index/translog/TranslogStats.java [hunk 1]
```diff
@@ -156,5 +184,17 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeVInt(uncommittedOperations);
         out.writeVLong(uncommittedSizeInBytes);
         out.writeVLong(earliestLastModifiedAge);
+        // TODO: if (out.getVersion().onOrAfter(Version.V_2_10_0)) {
+        if (out.getVersion().onOrAfter(Version.CURRENT)) {
```
Comment (ashking94) [inline server/src/main/java/org/opensearch/index/translog/TranslogStats.java:188]: lets set this to 2.10 here.

File: server/src/main/java/org/opensearch/index/translog/TranslogStats.java [hunk 2]
```diff
@@ -48,21 +49,33 @@
...
 
     public TranslogStats(StreamInput in) throws IOException {
         numberOfOperations = in.readVInt();
         translogSizeInBytes = in.readVLong();
         uncommittedOperations = in.readVInt();
         uncommittedSizeInBytes = in.readVLong();
         earliestLastModifiedAge = in.readVLong();
+        // TODO: remoteTranslogStats = in.getVersion().onOrAfter(Version.V_2_10_0) ? in.readOptionalWriteable(RemoteTranslogStats::new) :
+        // new RemoteTranslogStats();
+        remoteTranslogStats = in.getVersion().onOrAfter(Version.CURRENT)
```
Comment (ashking94) [inline server/src/main/java/org/opensearch/index/translog/TranslogStats.java:76]: lets set this to 2.10 here.

**Review by ashking94 — CHANGES_REQUESTED** [review]
Found that 2.10 branch has been cut already.

File: server/src/main/java/org/opensearch/index/translog/TranslogStats.java (hunk 1 above)
Comment (BhumikaSaini-Amazon) [reply server/src/main/java/org/opensearch/index/translog/TranslogStats.java]: done

File: server/src/main/java/org/opensearch/index/translog/TranslogStats.java (hunk 2 above)
Comment (BhumikaSaini-Amazon) [reply server/src/main/java/org/opensearch/index/translog/TranslogStats.java]: done

**Review by gbbafna — APPROVED** [review]

**Comment by gbbafna** [comment]
All tests passed . 

```
* What went wrong:
Execution failed for task ':distribution:bwc:bugfix:buildBwcLinuxTar'.
> Process '/var/jenkins/workspace/gradle-check/search/distribution/bwc/bugfix/build/bwc/checkout-2.9/gradlew [-Dbuild.snapshot=true, -Dscan.tag.NESTED, --parallel, distribution:archives:linux-tar:assemble, --build-cache]' finished with non-zero exit value 1
```

**Pull request merged**